packages/vite/src/node/plugins/html.ts

```diff
@@ -921,9 +923,20 @@ export function postImportMapHook(): IndexHtmlTransformHook {
       importMap = match
```

Comment:
Not sure if GitHub can highlight this correctly, but on line 919 we need to update the early-bail check to test preloads too. It's kinda an edge case, but may it would be simpler if we update the `moduleScriptRE` to handle both module scripts and module preload links.

Reply:
Yeah, merge to a single regex will simplify the logic. I think we can keep both `moduleScriptRE` and `modulePreloadLinkRE` declaration and join the two regex by `new RegExp(regexes.map(regex => regex.source).join("|"))` to make the regex easier to be understand.